client/blocks/simple-site-rename-form/index.jsx

```diff
@@ -103,24 +132,72 @@ export class SimpleSiteRenameForm extends Component {
 	};
 
 	onFieldChange = event => {
+		if ( this.props.isAvailabilityPending || this.props.isSiteRenameRequesting ) {
+			return;
+		}
+
 		const domainFieldValue = get( event, 'target.value', '' ).toLowerCase();
-		const shouldUpdateError = ! isEmpty( this.state.domainFieldError );
 
-		this.setState( {
-			domainFieldValue,
-			...( shouldUpdateError && {
-				domainFieldError: this.getDomainValidationMessage( domainFieldValue ),
-			} ),
-		} );
+		this.debouncedValidationCheck.cancel();
+		this.debouncedShowValidationMessage.cancel();
+
+		this.props.clearValidationError( this.props.siteId );
+		this.setState(
+			{
+				domainFieldValue,
+			},
+			this.setValidationState
+		);
 	};
 
+	debouncedShowValidationMessage = debounce( () => {
+		if ( this.state.validationMessage ) {
+			this.setState( {
+				showValidationMessage: true,
+			} );
+		}
+	}, VALIDATION_DEBOUNCE_MS );
+
+	debouncedValidationCheck = debounce( () => {
+		if ( ! isEmpty( this.state.domainFieldValue ) ) {
+			this.props.requestSiteAddressAvailability( this.props.siteId, this.state.domainFieldValue );
+		}
+	}, VALIDATION_DEBOUNCE_MS );
+
+	shouldShowValidationMessage() {
+		const { isAvailable, validationError } = this.props;
+		const { showValidationMessage } = this.state;
+		const serverValidationMessage = get( validationError, 'message' );
+
+		return isAvailable || showValidationMessage || !! serverValidationMessage;
+	}
+
+	getValidationMessage() {
+		const { isAvailable, validationError, translate } = this.props;
+		const { validationMessage } = this.state;
+		const serverValidationMessage = get( validationError, 'message' );
+
+		return isAvailable
+			? translate( 'Good news, this Site Address is available!' )
+			: validationMessage || serverValidationMessage;
+	}
+
 	render() {
-		const { currentDomain, currentDomainSuffix, isSiteRenameRequesting, translate } = this.props;
+		const {
+			currentDomain,
+			currentDomainSuffix,
+			isAvailabilityPending,
+			isAvailable,
+			isSiteRenameRequesting,
+			translate,
+		} = this.props;
+		const { domainFieldValue } = this.state;
 		const currentDomainName = get( currentDomain, 'name', '' );
 		const currentDomainPrefix = currentDomainName.replace( currentDomainSuffix, '' );
-		const { domainFieldError, domainFieldValue } = this.state;
-		const isDisabled =
-			! domainFieldValue || !! domainFieldError || domainFieldValue === currentDomainPrefix;
```

Comment:
Can we please keep the client side invalidation for when the entered text and current site address are the same?

With this change, it does a roundtrip & says:

> Sorry, that site address has already been used!

Reply:
Addressed here: https://github.com/Automattic/wp-calypso/pull/23492/files#diff-ac880d242a0b239e93d69cfa44e7981dR200

Thanks for spotting :)

Reply:
RE: This behavior, it was easier to PR it, so check out #24190 for some follow up here.